Apache-Beam-Jobs.md

```diff
@@ -227,6 +229,257 @@ error_pcoll = (
+
+Job names should follow the convention: `<Verb><Noun>Job`.
+
+-   For example:
+
+    ```python
+    class WeeklyDashboardStatsComputationJob(base_jobs.JobBase):
+        """BAD: Name does not begin with a verb."""
+
+        def run(self):
+            ...
+
+
+    class ComputeStatsJob(base_jobs.JobBase):
+        """BAD: Unclear what kind of stats are being computed."""
+
+        def run(self):
+            ...
+
+
+    class CountExplorationStatesJob(base_jobs.JobBase):
+        """GOOD: Name starts with a verb and "exploration states" is unambiguous."""
+
+        def run(self):
+            ...
+    ```
+
+Module names should follow the convention: `<noun>_<operation>_jobs.py`.
+
+-   For example:
+    * `blog_validation_jobs.py`
+    * `dashboard_stats_computation_jobs.py`
+    * `exploration_indexing_jobs.py`
+    * `exploration_stats_regeneration_jobs.py`
+    * `model_validation_jobs.py`
+
+    However, you should always prefer placing jobs in pre-existing modules if an appropriate one already exists.
+
+For this example, we will write our job in the module: `core/jobs/batch_jobs/exploration_inspection_jobs.py`.
+
+### 2. Override the `run()` method to operate on `self.pipeline`
+
+As illustrated in the Architecture section, jobs are organized by `Pipeline`s, `PTransform`s, and `PCollection`s. Thus, when writing your job, **you must build _off_ of your pipeline** in order to construct the DAG. We can represent this by adding a special `Pipeline` node signifying the _true_ beginning of the DAG:
+
+      ____________
+     /           /\
+    |  Pipeline |  |
+     \___________\/
+            |
+            | GetModels()
+            v
+    .--------------. Count states .--------. Sum .-------.
+    | Explorations | -----------> | Counts | --> | Total |
+    '--------------'              '--------'     '-------'
+
+Let's see how this would translate into code, starting with the Explorations.
+
+```python
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.platform import models
+
+(exp_models,) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class CountExplorationStatesJob(base_jobs.JobBase):
+
+    def run(self):
+        exp_model_pcoll = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExplorationModel.get_all(deleted=False))
+        )
+```
+
+Observe that:
+1. We're using `self.pipeline` as if it were a `PValue`
+2. We're using `ndb_io.GetModels` rather than `get_multi`
+3. We're passing a Query to `ndb_io.GetModels`
+
+It just so happens that `Pipeline` does indeed count as a `PValue`, with the special interface "`PBegin`". `PBegin` values signify that the value is never the output of a `PTransform`. Operations can branch off of `PBegin` values, like `ndb_io.GetModels()` does, to begin execution on a pipeline.
```

Comment:
"It just so happens" suggests coincidence/chance. Instead, try to explain how readers can think of Pipelines as PValues conceptually -- e.g. "Note that a `Pipeline` is a special case of a `PValue` that ..."

Reply:
Done.